### ISSUE TITLE
Regression with NavigationView now producing binding error when minimising it

### ISSUE THREAD

**Issue by beeradmoore** [microsoft/microsoft-ui-xaml#9384]
### Describe the bug

Updating my app to the latest WinUI version (1.5.240227000) I noticed I started getting a binding warning when I run it. The previous version (1.4.240211001) does not have this issue.

> Error: Converter failed to convert value of type 'Windows.Foundation.IReference`1<Microsoft.UI.Xaml.GridLength>' to type 'Double'; BindingExpression: Path='TemplateSettings.CompactPaneGridLength' DataItem='Microsoft.UI.Xaml.Controls.SplitView'; target element is 'Microsoft.UI.Xaml.Media.Animation.SplineDoubleKeyFrame' (Name='null'); target property is 'Value' (type 'Double'). 

I was able to keep deleting code until I got it to dissapear and I got an app layout down to a single `NavigationView`. 

```xml
<?xml version="1.0" encoding="utf-8"?>
<Window
    x:Class="NavViewBindingIssue.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:NavViewBindingIssue"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:muxc="using:Microsoft.UI.Xaml.Controls"
    mc:Ignorable="d">
    <muxc:NavigationView>

    </muxc:NavigationView>
</Window>
```

If you launch this there will be no binding issues. If you minmise the nav by clicking the hamburger button it will generate the above binding issue. 

Additionally if you make your nav view this it will show the binding error on launch.

```xml
    <muxc:NavigationView PaneDisplayMode="LeftCompact">
    
    </muxc:NavigationView>
```

### Steps to reproduce the bug

1. Create new WinUI project
2. Update nugets to latest versions
3. Change `MainWindow.xaml` to have the above `NavigationView`
4. (fix any issues of not compiling because you remove the normal button)
5. Run and toggle the side menu to see the binding error output on the debug console (and on screen helper bar)
6. Roll back one version to 1.4.240211001 and test again, binding issue is now gone.

### Expected behavior

No binding errors.

### Screenshots

N/A

### NuGet package version

WinUI 3 - Windows App SDK 1.5.0: 1.5.240227000

### Windows version

Windows 11 (23H2): Build 22631

### Additional context

N/A

**Comment by molesmoke** [comment]
I'm also seeing this if MAUI is built against WinUI 3 v1.5.240227000

**Comment by 27k1** [comment]
I get the same problem with WINUI 3 all latest nuGet packages. This issue is not closed, Please reopen.

**Comment by kmgallahan** [comment]
> Please reopen.

@27k1 It isn't closed.

**Comment by parko65** [comment]
Issue is still present in 1.5.240311000

**Comment by MPITech** [comment]
Confirmed, happens here as well still in 1.5.240311000.

**Comment by ranjeshj** [comment]
The offending piece is the makrup below in the SplitView template - we are trying to bind a gridlength to translatex - which are not compatible types.

<DoubleAnimationUsingKeyFrames Storyboard.TargetName="PaneClipRectangleTransform" Storyboard.TargetProperty="TranslateX">
      <SplineDoubleKeyFrame KeyTime="{StaticResource SplitViewPaneAnimationCloseDuration}" KeySpline="0.0,0.35 0.15,1.0" Value="{Binding TemplateSettings.CompactPaneGridLength, RelativeSource={RelativeSource Mode=TemplatedParent}}" />
  </DoubleAnimationUsingKeyFrames>

**Comment by kanyu** [comment]
Hi, I'm using 1.5.240428000 in WinUI3 and this problem has still happened. When does 1.6 come out? Thanks

**Comment by voroninp** [comment]
Seeing this error on MAUI when opening/closing Flyout:
> Error: Converter failed to convert value of type 'Windows.Foundation.IReference`1<Microsoft.UI.Xaml.GridLength>' to type 'Double'; BindingExpression: Path='TemplateSettings.CompactPaneGridLength' DataItem='Microsoft.UI.Xaml.Controls.SplitView'; target element is 'Microsoft.UI.Xaml.Media.Animation.SplineDoubleKeyFrame' (Name='null'); target property is 'Value' (type 'Double').

**Comment by parko65** [comment]
Has this issue been resolved, I'm no longer seeing any runtime binding errors. What, if any, update did this?

**Comment by beeradmoore** [comment]
I was working with 1.6 on the weekend, I don't recall seeing it anymore.

I'll check later though.

**Comment by Fred-Hudson-CST** [comment]
> Seeing this error on MAUI when opening/closing Flyout:
> 
> > Error: Converter failed to convert value of type 'Windows.Foundation.IReference`1<Microsoft.UI.Xaml.GridLength>' to type 'Double'; BindingExpression: Path='TemplateSettings.CompactPaneGridLength' DataItem='Microsoft.UI.Xaml.Controls.SplitView'; target element is 'Microsoft.UI.Xaml.Media.Animation.SplineDoubleKeyFrame' (Name='null'); target property is 'Value' (type 'Double').

I am also seeing this error in my MAUI project, did you find any workarounds

**Comment by beeradmoore** [comment]
@Fred-Hudson-CST , did you update the WindowsAppSDK with MAUI or just using what it comes with?

Looking at the NuGet package for the current stable [Microsoft.Maui.Core](https://www.nuget.org/packages/Microsoft.Maui.Core/8.0.91#dependencies-body-tab) (v8.0.91), under dependencies it looks like all variants of `net8.0-windows` have a dependency of Microsoft.WindowsAppSDK >= 1.5.240802000.

In theory installing the latest Microsoft.WindowsAppSDK nuget should fix this issue as it should have been fixed in v1.6. But there may be things break in MAUI because it is using a WindowsAppSDK that is newer than what its minimum dependency version is. Worst case scenario you add it and things break and you have to roll back and wait for MAUI team to bump the minimum dependency. Best case scenario everything works and you get the advantages of updated WindowsAppSDK version.

**Comment by Fred-Hudson-CST** [comment]
>In theory installing the latest Microsoft.WindowsAppSDK nuget should fix this issue as it should have been fixed in v1.6. But there may be things break in MAUI because it is using a WindowsAppSDK that is newer than what its minimum dependency version is. Worst case scenario you add it and things break and you have to roll back and wait for MAUI team to bump the minimum dependency. Best case scenario everything works and you get the advantages of updated WindowsAppSDK version.

Adding that did in fact break it, looks like ill just have to wait
